storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogSegmentLifecycleTest.java

```diff
@@ -197,333 +206,159 @@ public void testRemoteLogSegmentLifeCycle(RemoteLogSegmentLifecycleManager remot
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) {
-                return true;
-            }
-            if (o == null || getClass() != o.getClass()) {
-                return false;
-            }
-            EpochOffset that = (EpochOffset) o;
-            return epoch == that.epoch && offset == that.offset;
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(epoch, offset);
-        }
-
-        @Override
-        public String toString() {
-            return "EpochOffset{" +
-                    "epoch=" + epoch +
-                    ", offset=" + offset +
-                    '}';
-        }
-    }
-
-    private static Collection<Arguments> remoteLogSegmentLifecycleManagers() {
-        return Arrays.asList(Arguments.of(new RemoteLogMetadataCacheWrapper()),
-                             Arguments.of(new TopicBasedRemoteLogMetadataManagerWrapper()));
-    }
-
-    private void checkListSegments(RemoteLogSegmentLifecycleManager remoteLogSegmentLifecycleManager,
+    private void checkListSegments(RemoteLogMetadataManager metadataManager,
                                    int leaderEpoch,
-                                   RemoteLogSegmentMetadata expectedSegment)
+                                   RemoteLogSegmentMetadata expectedMetadata)
             throws RemoteStorageException {
         // cache.listRemoteLogSegments(leaderEpoch) should contain the above segment.
-        Iterator<RemoteLogSegmentMetadata> segmentsIter = remoteLogSegmentLifecycleManager.listRemoteLogSegments(leaderEpoch);
-        Assertions.assertTrue(segmentsIter.hasNext());
-        Assertions.assertEquals(expectedSegment, segmentsIter.next());
+        Iterator<RemoteLogSegmentMetadata> metadataIter =
+                metadataManager.listRemoteLogSegments(topicIdPartition, leaderEpoch);
+        assertTrue(metadataIter.hasNext());
+        assertEquals(expectedMetadata, metadataIter.next());
 
         // cache.listAllRemoteLogSegments() should contain the above segment.
-        Iterator<RemoteLogSegmentMetadata> allSegmentsIter = remoteLogSegmentLifecycleManager.listAllRemoteLogSegments();
-        Assertions.assertTrue(allSegmentsIter.hasNext());
-        Assertions.assertEquals(expectedSegment, allSegmentsIter.next());
+        Iterator<RemoteLogSegmentMetadata> allSegmentsIter = metadataManager.listRemoteLogSegments(topicIdPartition);
+        assertTrue(allSegmentsIter.hasNext());
+        assertEquals(expectedMetadata, allSegmentsIter.next());
     }
 
-    @ParameterizedTest(name = "remoteLogSegmentLifecycleManager = {0}")
-    @MethodSource("remoteLogSegmentLifecycleManagers")
-    public void testCacheSegmentWithCopySegmentStartedState(RemoteLogSegmentLifecycleManager remoteLogSegmentLifecycleManager) throws Exception {
-
-        try {
-            remoteLogSegmentLifecycleManager.initialize(topicIdPartition);
-
+    @ClusterTest(brokers = 3)
```

Comment:
It seems all test cases require `brokers = 3`. Maybe we can define it by `@ClusterTestDefaults(brokers = 3)`?